gunpowder/nodes/add_affinities.py

```diff
@@ -139,11 +142,15 @@ def process(self, batch, request):
         affinities_roi = request[self.affinities].roi
 
         logger.debug("computing ground-truth affinities from labels")
-        affinities = malis.seg_to_affgraph(
-                batch.arrays[self.labels].data.astype(np.int32),
-                self.affinity_neighborhood
-        ).astype(np.uint8)
 
+        # remove superfluous channel dim if it exists
+        arr = batch.arrays[self.labels].data.astype(np.int32)
+        if arr.shape[0] == 1:
+            arr.shape = arr.shape[1:]
```

Comment:
Not sure about removing the first channel here. I think I would expect the affinities to have the same shape as the labels.

Reply:
The idea was that labels might have shape `(1, z, y, x)` or  `(z, y, x)` but the function that computes the affinities expects `(z, y, x)`. If labels has shape `(z, y, x)` with `z` actually being `1` I guess we would still want to remove that dimension to use the 2d affinity function instead of the 3d one.

Reply:
That makes sense. Just not sure if that should be handled here or we would want to throw a error/warning and the user can remove a dimension upstream of `AddAffinites`.